php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -636,6 +647,28 @@ public DeclaredTypeTree DECLARED_TYPE() {
       b.firstOf(DNF_TYPE(), UNION_TYPE(), INTERSECTION_TYPE(), TYPE()));
   }
 
+  public PropertyHookListTree PROPERTY_HOOK_LIST() {
+    return b.<PropertyHookListTree>nonterminal(PHPLexicalGrammar.PROPERTY_HOOK_LIST).is(
+      f.propertyHookList(
+        b.token(LCURLYBRACE),
+        b.oneOrMore(PROPERTY_HOOK()),
+        b.token(RCURLYBRACE)));
+  }
+
+  public PropertyHookTree PROPERTY_HOOK() {
+    return b.<PropertyHookTree>nonterminal(PHPLexicalGrammar.PROPERTY_HOOK).is(
+      f.propertyHook(
+        b.zeroOrMore(ATTRIBUTE_GROUP()),
+        b.zeroOrMore(MEMBER_MODIFIER()),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        NAME_IDENTIFIER_OR_KEYWORD(),
```

Comment:
I tried to do it by adding it to `PHPKeywords`but this will conflict the parser when other method declarations try to declare `get`or `set`methods.
If you know another way, I'm happy to change it

Reply:
Found one!